src/org/labkey/targetedms/parser/PrecursorChromInfo.java

```diff
@@ -450,26 +458,54 @@ public Chromatogram createChromatogram(TargetedMSRun run)
                     if (skydData != null)
                     {
                         Path skydPath = skydData.getFilePath();
-                        LOG.debug("Attempting to fetch chromatogram bytes (possibly cached) from " + skydPath + " for PrecursorChromInfo " + _generalMoleculeChromInfoId);
-                        byte[] onDemandBytes = ON_DEMAND_CHROM_CACHE.get(new Tuple3<>(skydPath, _chromatogramOffset, _chromatogramLength));
-                        if (databaseBytes != null && !Arrays.equals(databaseBytes, onDemandBytes))
+                        if (skydPath == null)
                         {
-                            LOG.error("Chromatogram bytes for PrecursorChromInfo " + _generalMoleculeChromInfoId + " do not match between .skyd and DB. Using database copy. Lengths: " + (onDemandBytes == null ? "null" : Integer.toString(onDemandBytes.length)) + " vs " + databaseBytes.length);
+                            status = Chromatogram.SourceStatus.skydMissing;
+                            LOG.debug("No path available for PrecursorChromInfo " + _generalMoleculeChromInfoId + ", bucket may be unavailable for URL " + skydData.getDataFileUrl());
                         }
                         else
                         {
-                            compressedBytes = onDemandBytes;
+                            LOG.debug("Attempting to fetch chromatogram bytes (possibly cached) from " + skydPath + " for PrecursorChromInfo " + _generalMoleculeChromInfoId);
+                            byte[] diskBytes = ON_DEMAND_CHROM_CACHE.get(new Tuple3<>(skydPath, _chromatogramOffset, _chromatogramLength));
+                            if (diskBytes == null)
+                            {
+                                status = Chromatogram.SourceStatus.skydMissing;
+                            }
+                            else if (!Arrays.equals(databaseBytes, diskBytes))
+                            {
+                                LOG.error("Chromatogram bytes for PrecursorChromInfo " + _generalMoleculeChromInfoId + " do not match between .skyd and DB. Using database copy. Lengths: " + (diskBytes == null ? "null" : Integer.toString(diskBytes.length)) + " vs " + databaseBytes.length);
```

Comment:
diskBytes == null is always null here.

Reply:
True, and databaseBytes could have been null. Update pushed.

Reply:
Looks good, thanks.